### DISCUSSION TITLE
Issue with preinstall script

### DISCUSSION THREAD

**Discussion by lainermeister** [community/community#145149] [npm]
Hi!

I am trying to add an auth script that should run to get authentication to an internal package registry (in Google Artifact Registry). I would ideally like the script to run before `npm install`, so I put it in a `preinstall` script. However, it doesn't seem like it runs before npm checks if we have permissions to install the package.

Here is a video with more details:
https://drive.google.com/file/d/1Wax_g3HYpvKtHUI7f6rq75mr5bIN8Qvf/view?usp=sharing

Is this a known issue?

Thanks in advance for your help!
Elaine

**Comment by dalvarez2596** [comment]
Hi Elaine,
Yeah actually, there are still a lot of discussions open about that. Some people agree that the preinstall behavior should be, as you mentioned, to run the preinstall before installing the dependencies, but that's not the case. You can refer to these discussions for more information:
This is a PR that was mentioned regarding that issue, which was closed because it would be a breaking change: [#2713](https://github.com/npm/cli/pull/2713).

Also here are some discussions on the topic:
[#2660](https://github.com/npm/cli/issues/2660)
[#2883](https://github.com/npm/cli/issues/2883)

Hope this works :D

**Reply by lainermeister** [reply]
@dalvarez2596 Thanks for your quick response! Do you have suggestions on path forward here?

**Comment by denis23x** [comment]
Hi @lainermeister 

To be honest, I don't know why the preinstall isn't called in your case, but I can definitely tell you that you can automate your processes using a bash script 😊

Please take a look here https://www.geeksforgeeks.org/bash-scripting-introduction-to-bash-and-bash-scripting/

Bash scripts are widely used in development, especially when more powerful functionality is needed than what npm hooks can provide, so feel free to use it.

If you prefer JavaScript, here is another great option https://github.com/google/zx
